serverlessworkflow/modules/ROOT/pages/job-services/core-concepts.adoc

```diff
@@ -1,107 +1,584 @@
+* `Recipient`: the entity that is called on the job execution for the given interaction mode, and  receives the execution parameters.
 
-All timer nodes that might be declared in a workflow, are handled by the job service, which is responsible for managing, scheduling, and firing all actions (jobs) to be executed in the workflows.
+image::job-services/Job-Service-Generic-Diagram.png[]
+
+[#integration-with-the-workflows]
+== Integration with the Workflows
+
+In the context of the {product_name}s, the Job Service is responsible for controlling the execution of the time-triggered actions. And thus, all the time-base states that you can use in a workflow, are handled by the interaction between the workflow and the Job Service.
+
+For example, every time the workflow execution reaches a state with a configured timeout, a corresponding job is created in the Job Service, and when the timeout is met, a http callback is executed to notify the workflow.
+
+image::job-services/Time-Based-States-And-Job-Service-Interaction.png[]
+
+To set up this integration you can use different xref:job-services/quarkus-extensions.adoc#job-service-quarkus-extensions[communication alternatives], that must be configured by combining the Job Service and the Quarkus Workflow Project configurations.
 
 [NOTE]
 ====
-Suppose the workflow service is not configured to use the Job Service or there is no such service running.
-In that case, all timer-related actions use an embedded in-memory implementation of the Job Service, which should not be used in production, since when the application shutdown, all timers are lost, which in a serverless architecture is a very common behavior with the scale to zero approach.
-That said, the usage of in-memory Job Service can only be done for testing or development, but not for production.
+If the project is not configured to use the Job Service, all time-based actions will use an in-memory implementation of that service.
+However, this setup must not be used in production, since every time the application is restarted, all the timers are lost.
+This last is not suited for serverless architectures, where the applications might scale to zero at any time, etc.
 ====
 
-The main goal of the Job Service is to work with only active jobs. The Job Service tracks only the jobs that are scheduled and that need to be executed. When a job reaches a final state, the job is removed from the Job Service.
+== Jobs life-span
 
-When configured in your environment, all the jobs information and status changes are sent to the {product_name} `Data
+Since the main goal of the Job Service is to work with the active jobs, such as the scheduled jobs that needs to be executed, when a job reaches a final state, it is removed from the Job Service.
+However, in cases where you want to keep the information about the jobs in a permanent repository, you can configure the Job Service to produce status change events, that can be collected by the {product_name} `Data
```

Comment:
```suggestion
However, in some cases where you want to keep the information about the jobs in a permanent repository, you can configure the Job Service to produce status change events, that can be collected by the {product_name} `Data
```

Reply:
Are we introducing here the Kogito Data Index ?
Do we have a link to some documentation ? Else the user will be confused and won't know where to look ...